kythe/cxx/indexer/cxx/recursive_type_visitor.h

```diff
@@ -232,6 +232,16 @@ DEF_TRAVERSE_TYPEPAIR(DependentVectorType, {
 DEF_TRAVERSE_TYPEPAIR(ExtVectorType, {
   return getDerived().TraverseType(TL.getTypePtr()->getElementType());
 });
+DEF_TRAVERSE_TYPEPAIR(ConstantMatrixType, {
+  return getDerived().TraverseType(TL.getTypePtr()->getElementType());
+});
+DEF_TRAVERSE_TYPEPAIR(DependentSizedMatrixType, {
+  if (T->getRowExpr())
```

Comment:
Does this mutate state in T? And that's why you have the same if block below?

Reply:
This was taken from the internal change and it was wrong.  Cases like this should probably just delegate to `TraverseTypeLoc`, but 🤷‍♂️

Reply:
(Pushing the fix now)